### PR TITLE
Added pivot button for intro page

### PR DESCRIPTION
**Pull request by meanwhile7**
- **Issue Ticket Number:** #367

**Testing being written separately**

-  Open issue #441 

#### Intro Button:
Screenshot from 2023-07-05 21-18-52
* Only superuser has access to the button 
* Button visible to users whose intro page is available
* Locked for Non-super User
#### Non superuser see the button disabled with a lock icon and popover describing missing permissions preview
[popover preview.webm](https://github.com/Real-Dev-Squad/website-dashboard/assets/48379860/e67766e0-3030-4e5f-b640-70f5e660351b)

### CODE REVIEW

File: users/details/style.css [hunk 1]
```diff
@@ -140,6 +140,25 @@ li {
   height: 25px;
 }
 
+.enabled {
+  background-color: var(--loader-gray-color);
+  color: black;
```
Comment (sahsisunny) [inline users/details/style.css:145]: Please make variables and then use it

File: users/details/style.css [hunk 2]
```diff
@@ -140,6 +140,25 @@ li {
...
+  cursor: pointer;
+}
+
+.disabled {
+  position: relative;
+  background-color: #dddddd;
+  color: #999999;
```
Comment (sahsisunny) [inline users/details/style.css:152]: Same as above create variable for the colors

Comment (meanwhile7) [reply users/details/style.css:152]: why everything works fine in ui part I found no issue

File: users/details/script.js [hunk 3]
```diff
@@ -108,6 +108,44 @@ function generateUserData(userData) {
     type: 'div',
     classList: ['user-details-main'],
   });
+  const socials = generateSocialMediaLinksList();
+
+  if (userData?.profileURL) {
+    checkUserIsSuperUser().then((isSuperUser) => {
+      console.log(isSuperUser);
```
Comment (prakashchoudhary07) [inline users/details/script.js:115]: remove console logs

Comment (meanwhile7) [reply users/details/script.js]: made necessary changes

File: users/details/style.css (hunk 2 above)
Comment (meanwhile7) [reply users/details/style.css]: made necessary changes in latest commit

**Review by kotesh-arya — DISMISSED** [review]

**Comment by kotesh-arya** [comment]
Please check this out.
image

File: users/details/script.js [hunk 4]
```diff
@@ -108,6 +108,43 @@ function generateUserData(userData) {
     type: 'div',
     classList: ['user-details-main'],
   });
+  const socials = generateSocialMediaLinksList();
+
+  if (userData?.profileURL) {
```
Comment (iamitprakash) [inline users/details/script.js:113]: what if userData.profileURL property is undefined

Comment (meanwhile7) [reply users/details/script.js:113]: this button is not created for every user only for those whose intro page exist that's why this condition exist
ankush requested I want option of the button for users whose intro page exist 
that's why if its undefined it means intro page for that user don't exist and button will not be created for that user

**Review by iamitprakash — CHANGES_REQUESTED** [review]
add test case stats

File: __tests__/users/user-management-home-screen.test.js [hunk 5]
```diff
@@ -80,6 +81,11 @@ describe('Tests the User Management User Listing Screen', () => {
     await browser.close();
   });
 
+  it('Checks the pivot button', async () => {
+    pivot = page.waitForSelector('.Intro');
+    expect(pivot).toBeTruthy();
+  });
+
```
Comment (prakashchoudhary07) [inline __tests__/users/user-management-home-screen.test.js:88]: Can we few more test case, this just check if its present

**Review by iamitprakash — APPROVED** [review]

**Review by iamitprakash — APPROVED** [review]

File: utils.js [hunk 6]
```diff
@@ -16,7 +16,7 @@ async function getSelfUser() {
 
 async function checkUserIsSuperUser() {
   const self_user = await getSelfUser();
-  return self_user?.roles['super_user'];
+  return self_user?.roles['super_user'] || false;
 }
```
Comment (Pratiyushkumar) [inline utils.js:20]: why are we returning false from here?

File: __tests__/users/user-management-home-screen.test.js
Comment (Pratiyushkumar) [inline __tests__/users/user-management-home-screen.test.js:1]: test for if there is a super user or not, if user is super user then what should happen and if not then what should happen

File: utils.js (hunk 6 above)
Comment (meanwhile7) [reply utils.js:20]: my bad I will undo this change i just thought instead of returning undefined let it return false would look better.

**Review by shubham-y — APPROVED** [review]

**Review by Achintya-Chatterjee — APPROVED** [review]
LGTM

**Review by Achintya-Chatterjee — APPROVED** [review]
LGTM

**Review by Pratiyushkumar — APPROVED** [review]

**Review by sahsisunny — APPROVED** [review]

**Pull request merged**